File: player/src/main/java/com/example/player/MainActivity.java
package com.example.player;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.Manifest;
import android.opengl.GLSurfaceView;
import android.os.Bundle;
import android.view.SurfaceHolder;
import android.view.SurfaceView;

import com.mylibrary.activity.PermissionsActivity;

import java.io.IOException;
import java.io.InputStream;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        PermissionsActivity.startActivityForResult(this, 1,
                new String[]{
                        Manifest.permission.WRITE_EXTERNAL_STORAGE,
                });
        SurfaceView surfaceView = findViewById(R.id.playView);
        surfaceView.getHolder().addCallback(new SurfaceHolder.Callback() {
            @Override
            public void surfaceCreated(@NonNull SurfaceHolder holder) {
                try {
                    InputStream inputStream = getResources().getAssets().open("out.h264");
                    H264Codec codec = new H264Codec(holder.getSurface(), H264Codec.toByteArray(inputStream));
                    codec.start();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }

            @Override
            public void surfaceChanged(@NonNull SurfaceHolder holder, int format, int width, int height) {

            }

            @Override
            public void surfaceDestroyed(@NonNull SurfaceHolder holder) {

            }
        });
    }
}

File: MyJetpack/src/main/java/com/myjetpack/MainActivity.kt
package com.myjetpack

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.LifecycleRegistry

//activity实现接口LifecycleOwner后，可以添加生命周期观察者。
class MainActivity : AppCompatActivity(), LifecycleOwner {
    companion object {
        const val TAG = "MainActivity";
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        lifecycle.addObserver(MyObserve(lifecycle))
        Log.e(Companion.TAG, "get savedInstanceState value = " + savedInstanceState?.getString("save"))
    }

    override fun onPause() {
        super.onPause()
        Log.e(Companion.TAG, "onPause");
    }

    override fun onResume() {
        super.onResume()
        Log.e(Companion.TAG, "onResume");
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.e(Companion.TAG, "onDestroy");
    }

    override fun onSaveInstanceState(outState: Bundle, outPersistentState: PersistableBundle) {
        super.onSaveInstanceState(outState, outPersistentState)
        Log.e(Companion.TAG, "excute onSaveInstanceState");
        outState.putString("save", "123")
    }

}